tiledb/sm/query/writer.cc

```diff
@@ -1304,9 +1309,14 @@ Status Writer::check_subarray() const {
     if (layout_ == Layout::UNORDERED && subarray_.is_set())
       return LOG_STATUS(Status::WriterError(
           "Cannot initialize query; Setting a subarray in unordered writes for "
-          "dense arrays in inapplicable"));
+          "dense arrays is inapplicable"));
+  } else {  // state is !array_schema_->dense()
+    if (subarray_.is_set())
+      // Note: previously handled in Writer::add_range()
```

Comment:
I believe that line is necessary to preserve semantic equivalence (in Subarray::add_range()) with what Writer::add_range() does.
Without it, a sparse query with a subarray_.is_set()  will be treated as Status::Ok(), 
whereas in the query-subarray based (add_range) paths it will be treated as an error.
Regarding maintenance issues, practically speaking they are going to be unavoidable while 
you have similar functionality paths via tiledb_query_etc and tiledb_subarray_etc.
Any time logic changes in one path the other is going to have to be examined for equivalent changes (which are not always readily apparent, with the old paths splitting various audits between Query and its internal reader/writer entities, while Subarray does not have that same structure since it is settable independently and then may be set into a Query, so the two approaches are somewhat inverted from each other.)

Reply:
Should be fine. Remove the note maybe?

Reply:
Removed.